cilium-cli/connectivity/tests/encryption_v2.go

```diff
@@ -0,0 +1,592 @@
+	dstWord3 := fmt.Sprintf("0x%02x%02x%02x%02x", dstBytes[8], dstBytes[9], dstBytes[10], dstBytes[11])
+	dstWord4 := fmt.Sprintf("0x%02x%02x%02x%02x", dstBytes[12], dstBytes[13], dstBytes[14], dstBytes[15])
+
+	clientInnerIPv6Src := fmt.Sprintf(innerIPv6Src, srcWord1, srcWord2, srcWord3, srcWord4)
+	clientInnerIPv6Dst := fmt.Sprintf(innerIPv6Dst, dstWord1, dstWord2, dstWord3, dstWord4)
+
+	serverInnerIPv6Src := fmt.Sprintf(innerIPv6Src, dstWord1, dstWord2, dstWord3, dstWord4)
+	serverInnerIPv6Dst := fmt.Sprintf(innerIPv6Dst, srcWord1, srcWord2, srcWord3, srcWord4)
+
+	clientFilter = fmt.Sprintf(fmtFilter, port, clientInnerIPv6Src, clientInnerIPv6Dst)
+	serverFilter = fmt.Sprintf(fmtFilter, port, serverInnerIPv6Dst, serverInnerIPv6Src)
+	return clientFilter, serverFilter, nil
+}
+
+func (s *podToPodEncryptionV2) nativeTCPDumpFilters6(ctx context.Context) (clientFilter string, serverFilter string, err error) {
+	if ctx.Err() != nil {
+		return "", "", fmt.Errorf("Context already cancelled")
+	}
+
+	fmtNativeFilter := "src %s and dst %s"
+
+	src, err := netip.ParseAddr(s.client.Address(features.IPFamilyV6))
+	if err != nil {
+		return "", "", fmt.Errorf("Failed to parse client pod IP: %w", err)
+	}
+	dst, err := netip.ParseAddr(s.server.Address(features.IPFamilyV6))
+	if err != nil {
+		return "", "", fmt.Errorf("Failed to parse server pod IP: %w", err)
+	}
+
+	clientFilter = fmt.Sprintf(fmtNativeFilter, src, dst)
+	serverFilter = fmt.Sprintf(fmtNativeFilter, dst, src)
+	return clientFilter, serverFilter, nil
+}
+
+// resolveTCPDumpFilters6 is the analog of resolveTCPDumpFilters4 but for IPv6.
+func (s *podToPodEncryptionV2) resolveTCPDumpFilters6(ctx context.Context) (clientFilter string, serverFilter string, err error) {
+	if ctx.Err() != nil {
+		return "", "", fmt.Errorf("Context already cancelled")
+	}
+
+	if s.tunnelMode.Enabled {
+		return s.tunnelTCPDumpFilters6(ctx)
+	}
+
+	return s.nativeTCPDumpFilters6(ctx)
+}
+
+// startSniffers will start TCPdump on both the client and the server pod's host
+// namespaces.
+//
+// if IPv6 is enabled for the cluster IPv6 specific sniffers will be started
+// as well.
+//
+// if encryption is enabled we will put the sniffer into Assert mode where any
+// captured packets indicates a test failure.
+//
+// conversely if encryption is disabled the sniffer is placed into Sanity mode
+// where packets are expected. this is useful, in a rather indirect way,
+// to prove that the generated tcpdump filters are working correctly and
+// capturing the traffic traffic which would be a leak if encryption was enabled.
+func (s *podToPodEncryptionV2) startSniffers(ctx context.Context, t *check.Test) error {
+	if ctx.Err() != nil {
+		return fmt.Errorf("Context already cancelled")
+	}
+
+	mode := sniff.ModeSanity
+	if s.encryptMode.Enabled {
+		t.Infof("Encryption is enabled: test will fail if plain-text packets are seen.")
```

Comment:
Would change `t.Info` to `t.Debug` where possible (here's just one example), to keep the logs when running CLI test with `-d`. This also aligns with our old tests.

```suggestion
		t.Debug("Encryption is enabled: test will fail if plain-text packets are seen.")
```